src/foam/u2/detail/SectionView.js

```diff
@@ -38,37 +55,74 @@ foam.CLASS({
       class: 'Boolean',
       name: 'showTitle',
       value: true
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.core.ArraySlot',
+      name: 'visibilitySlots_',
+      documentation: `
+        An array of slots used to control the visibility of the child
+        SectionedDetailPropertyViews of this view. Used internally to figure out
+        which children are the first and last ones visible so we can adjust the
+        padding on those elements to make sure nested
+        SectionedDetailPropertyViews don't cause unnecessary padding.
+      `,
+      factory: function() {
+        return this.ArraySlot.create();
+      }
+    },
+    {
+      class: 'Int',
+      name: 'firstVisibleIndex_',
+      expression: function(visibilitySlots_) {
+        return visibilitySlots_.get().indexOf(true);
+      }
+    },
+    {
+      class: 'Int',
+      name: 'lastVisibleIndex_',
+      expression: function(visibilitySlots_) {
+        return visibilitySlots_.get().lastIndexOf(true);
+      }
     }
   ],
   methods: [
     function initE() {
       var self = this;
-      this.addClass(this.myClass());
+      var slots = [];
```

Comment:
Hmm. This is going to grow larger every time section, showTitle, or section$title changes. You probably just want to declare this on line 98 and then set the ArraySlot just before returning.

Reply:
Resolved by 19661fbb0796bd0e1cc90ec1b74d1e9690800f03.